src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,62 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_LENGTH = 6;
+    private static final int MINIMAL_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RuntimeException {
+        if (user == null) {
+            throw new InvalidUserException("User can't be null!");
+        }
+        StringBuilder exception = new StringBuilder();
+        if (!checkUserForAge(user)) {
+            exception.append("Incorrect age!").append(System.lineSeparator());
+        }
+        if (!checkUserLoginForIdentity(user)) {
+            exception.append("There is user with this login already!")
+                    .append(System.lineSeparator());
+        }
+        if (!checkUserLoginForIdentity(user) || !checkUserLoginForLength(user)) {
+            exception.append("Incorrect login!").append(System.lineSeparator());
+        }
+        if (!checkUserPasswordForLength(user)) {
+            exception.append("Incorrect password!").append(System.lineSeparator());
+        }
+        if (exception.length() == 0) {
+            storageDao.add(user);
+            return user;
+        }
+        throw new InvalidUserException(exception.toString());
```

Comment:
well if you opt to do it this way, then to make it better: Create a list of string     
List<String> errorMessages = new ArrayList<>();
then add messages one by one
then         throw new InvalidUserException(String.join(System.lineSeparator(), errorMessages));

Reply:
Can you explain please why this method is better?